src/ImageLoaders/IHex32/IntelHex32Loader.cs

```diff
@@ -0,0 +1,242 @@
+
+            /// <summary>
+            /// Adds a series of data byte to the memory chunks list as decoded from an IHex32 data record.
+            /// </summary>
+            /// <remarks>
+            /// The current implementation assumes that each IHex32 data record loads unique data block (no overlap).
+            /// </remarks>
+            /// <param name="address">The address from the decoded IHex32 record.</param>
+            /// <param name="data">The binary bytes contained in the IHex32 record.</param>
+            public void AddData(uint address, byte[] data)
+            {
+                if (data == null || data.Length < 1)
+                    return;
+
+                if (currAddr == null)
+                {
+                    currAddr = Address.Ptr32(address);
+                    currMemChunk = new MemChunk(currAddr);
+                    memChunks.Add(currAddr, currMemChunk);
+                }
+                else
+                {
+                    currAddr = Address.Ptr32(address);
+                    if (nextAddr != currAddr)
+                    {
+                        currMemChunk = GetPredChunk(currAddr);
+                        if (currMemChunk == null)
+                        {
+                            currMemChunk = new MemChunk(currAddr);
+                            memChunks.Add(currAddr, currMemChunk);
+                        }
+                    }
+                }
+
+                currMemChunk.Datum.AddRange(data);
+                nextAddr = currAddr + data.Length;
+            }
+
+            public IEnumerator<MemChunk> GetEnumerator() => memChunks.Values.GetEnumerator();
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        #endregion
+
+        #region Locals
+
+        private DecompilerEventListener listener;
+
+        #endregion
+
+        #region Constructors
+
+        public IntelHex32Loader(IServiceProvider services, string filename, byte[] imgRaw)
+            : base(services, filename, imgRaw)
+        {
+        }
+
+        #endregion
+
+        #region ImageLoader interface implementation
+
+        /// <summary>
+        /// If nothing else is specified, this is the address at which the image will be loaded.
+        /// </summary>
+        /// <value>
+        /// The preferred base address.
+        /// </value>
+        public override Address PreferredBaseAddress { get; set; } = Address32.NULL;
+
+        /// <summary>
+        /// Loads the image into memory starting at the specified address.
+        /// Not used for Intel Hex as this format contains no meta-data.
+        /// </summary>
+        /// <param name="addrLoad">Base address of program image. IGNORED.</param>
+        /// <returns>
+        /// A <see cref="Program"/> instance.
+        /// </returns>
+        public override Program Load(Address addrLoad) => throw new NotImplementedException();
```

Comment:
The correct exception type is `NotSupportedException`. `NotImplementedException` implies that we haven't got around to writing any code yet.